JustSaying/JustSayingFluently.cs

```diff
@@ -231,16 +231,18 @@ public IFluentSubscription IntoQueue(string queuename)
                 : TopicHandler<T>();
 
             Bus.SerialisationRegister.AddSerialiser<T>(_serialisationFactory.GetSerialiser<T>());
-            var handlers = handlerResolver.ResolveHandlers<T>().ToList();
-            if (!handlers.Any())
+            if(!handlerResolver.ResolveHandlers<T>().Any())
             {
                 throw new HandlerNotRegisteredWithContainerException(string.Format("IHandler<{0}> is not regsistered in the container.", typeof(T).Name));
             }
-            foreach (var handler in handlers)
+            if (handlerResolver.ResolveHandlers<T>().Count() > 1)
             {
-                Bus.AddMessageHandler(handler);
-                Log.Info(string.Format("Added a message handler - Topic: {0}, QueueName: {1}, HandlerName: {2}", _subscriptionConfig.Topic, _subscriptionConfig.QueueName, handler.GetType().Name));
+                throw new NotSupportedException(string.Format("There are more than one registration for IHandler<{0}>. JustSaying currently does not support multiple registration for IHandler<T>.", typeof(T).Name));

```

Comment:
Allowing multiple registration requires a lot of thinking and perhaps major redesign. This can be done later.